<?xml version='1.0'?>

<launch>

    <!-- global variables -->
    <arg name='car_name' default='car_1' />
    <arg name='listen_offboard' default='true' />
    <arg name='remote_control' default='true'/>

    <!-- launch rviz with mapping configuration -->
    <node name='remote_visualization' pkg='rviz' type='rviz' args='-d $(find f1tenth-sim)/rviz/$(arg car_name).rviz' />

    <group ns='$(arg car_name)'>
        <!-- joy_node using F310/F710 -->
        <group if='$(arg remote_control)'>
            <node name='joy_node' pkg='joy' type='joy_node'>
                <param name='autorepeat_rate' value='20.0' />
	        </node> 
        </group>
        <!-- keyboard teleop -->
        <group unless='$(arg remote_control)'>
            <node name = 'keyboard_teleop' pkg = 'f1tenth-sim' type = 'keyboard_teleop.py' 
                args = '$(arg car_name)' output = 'screen'/>
        </group>

        <!-- bring up command multiplexer -->
        <node name='command_multiplexer' pkg='f1tenth-sim' type='command_multiplexer.py' args='$(arg car_name)
                                 $(arg listen_offboard) $(arg remote_control)' output='screen' />

    </group>

</launch>
